modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormAdminDisplayContextTest.java

```diff
@@ -160,6 +163,47 @@ public void testGetSharedFormURLForSharedFormInstance() throws Exception {
 			getSharedFormURL(), _ddmFormAdminDisplayContext.getSharedFormURL());
 	}
 
+	@Test
+	public void testIsFormExpired() throws Exception {
+		Assert.assertFalse(_ddmFormAdminDisplayContext.isFormExpired());
+
+		DDMFormInstanceSettings ddmFormInstanceSettings = mock(
+			DDMFormInstanceSettings.class);
+
+		DDMFormInstanceRecord ddmFormInstanceRecord =
+			_mockDDMFormInstanceRecord(ddmFormInstanceSettings);
+
+		when(
+			_httpServletRequest.getAttribute(
+				Matchers.eq(
+					DDMFormWebKeys.DYNAMIC_DATA_MAPPING_FORM_INSTANCE_RECORD))
+		).thenReturn(
+			ddmFormInstanceRecord
+		);
+
+		when(
+			ddmFormInstanceSettings.neverExpire()
+		).thenReturn(
+			true
+		);
+
+		Assert.assertFalse(_ddmFormAdminDisplayContext.isFormExpired());
+
+		when(
+			ddmFormInstanceSettings.expirationDate()
+		).thenReturn(
+			"1987-09-22"
+		);
+
+		when(
+			ddmFormInstanceSettings.neverExpire()
+		).thenReturn(
+			false
+		);
+
+		Assert.assertTrue(_ddmFormAdminDisplayContext.isFormExpired());
+	}
```

Comment:
@rodrigopaulino, I see that this unit test checks 3 test cases, in light of this how about split it into 3 unit tests? e.g. a unit test for each test case. For example:

- testFormNotExpiredWithInexistentForm
- testFormNotExpiredWithNeverExpireSetting
- testFormExpired

It's just a suggestion. :slightly_smiling_face: Feel free to use it or not. :wink: